Remove CUDNN_BATCHNORM_SPATIAL_PERSISTENT mode for accuracy

xw285cornell opened the pull request:
Summary:
In S163230, we've found CuDNN 7 upgrade causes accuracy drop in training convolution network such as ResNeXt-101 (~0% accuracy), and video R(2+1)D (65 --> 63%).

We've fixed this in Caffe2 D9601217, and we should do the same to ATen as well.

Differential Revision: D13025486

soumith commented on aten/src/ATen/native/cudnn/BatchNorm.cpp:
```diff
@@ -88,10 +88,10 @@ std::tuple<Tensor, Tensor, Tensor> cudnn_batch_norm(
-    if(training)
-      mode = CUDNN_BATCHNORM_SPATIAL_PERSISTENT;
-#endif
+    // TODO(T31829456): The new CUDNN_BATCHNORM_SPATIAL_PERSISTENT mode was
```
can you remove reference to internal task T31829456

xw285cornell replied:
Sorry didn't see this yesterday, removing

ezyang commented:
CC @csarofeen

ezyang reviewed (approved):
approved with comments

xw285cornell committed:
Remove CUDNN_BATCHNORM_SPATIAL_PERSISTENT mode for accuracy (#13844)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13844

In S163230, we've found CuDNN 7 upgrade causes accuracy drop in training convolution network such as ResNeXt-101 (~0% accuracy), and video R(2+1)D (65 --> 63%).

We've fixed this in Caffe2 D9601217, and we should do the same to ATen as well.

Reviewed By: ezyang

Differential Revision: D13025486

fbshipit-source-id: fe54180f0a97e4b759ad81b9b2bbffde4aa33047
